File: CS/LoadMissingFonts/Form1.cs
using DevExpress.Drawing;
using DevExpress.XtraReports.UI;
using FontDemoReport;

namespace LoadMissingFonts {
    public partial class Form1 : Form {
        public Form1() {            
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            using (var report = new DemoReport()) {
                DXFontRepository.QueryNotFoundFont += Report_QueryNotFoundFont;
                using (var tool = new ReportPrintTool(report)) {
                    tool.ShowRibbonPreviewDialog();
                }
            }
            Close();
        }

        private static void Report_QueryNotFoundFont(object sender, NotFoundFontEventArgs e) {
            var service = new FontCollectorService();
            var fontData = service.ProcessFont(e.RequestedFont).Result;
            e.FontFileData = fontData;
        }

      
    }
}

File: VB/LoadMissingFonts/Form1.vb
Imports DevExpress.Drawing
Imports DevExpress.XtraReports.UI
Imports FontDemoReport

Namespace LoadMissingFonts

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Using report = New DemoReport()
                DevExpress.Drawing.DXFontRepository.QueryNotFoundFont += AddressOf Report_QueryNotFoundFont
                Using tool = New ReportPrintTool(report)
                    tool.ShowRibbonPreviewDialog()
                End Using
            End Using

            Close()
        End Sub

        Private Shared Sub Report_QueryNotFoundFont(ByVal sender As Object, ByVal e As NotFoundFontEventArgs)
            Dim service = New FontCollectorService()
            Dim fontData = service.ProcessFont(e.RequestedFont).Result
            e.FontFileData = fontData
        End Sub
    End Class
End Namespace
